### ISSUE TITLE
Intermitent TLS / SSL errors on macOS

### ISSUE THREAD

**Issue by itamarst** [eventlet/eventlet#997]
CI has flaky tests on macOS runners, always in SSL tests, e.g. https://github.com/eventlet/eventlet/actions/runs/11934616913/job/33264113200?pr=996 but I've seen different errors like "invalid data" SSL errors.

**Comment by 4383** [comment]
Indeed, we had a parallel discussion days ago concerning the MacOS jobs https://github.com/eventlet/eventlet/pull/988#issuecomment-2476464828

**Comment by 4383** [comment]
However, I didn't checked if the errors are the same than the one you highlighted here
